src/legacy/ui/public/visualize/loader/embedded_visualize_handler.ts

```diff
@@ -435,7 +436,13 @@ export class EmbeddedVisualizeHandler {
     this.fetchAndRender();
   };
 
+  private cancel = () => {
+    this.dataLoader.cancel();
+  };
+
   private fetch = (forceFetch: boolean = false) => {
+    this.cancel();
```

Comment:
it doesn't need to do that here, we can do it internally in visualize loader (embedded_visualize_handler)